cumulusci/tasks/salesforce/promote_2gp_package.py

```diff
@@ -0,0 +1,308 @@
+from typing import List, Dict, Optional
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+from cumulusci.core.config.util import get_devhub_config
+from cumulusci.core.exceptions import CumulusCIException, TaskOptionsError
+from cumulusci.salesforce_api.utils import get_simple_salesforce_connection
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class Promote2gpPackageVersion(BaseSalesforceApiTask):
+    """
+    Promote a 2GP Package version from a Beta to a Production release.
+
+    The terms "promoted" and "released" are the same in the context of
+    a 2GP package. Once "promoted" a package is considered to be "released".
+    Hence, the "promotion" of a package is just an update to the
+    Package2Version.IsReleased field.
+
+    The abbreviation `spv` in variable names is shorthand for "SubscriberPackageVersion".
+    """
+
+    task_docs = """Promote a Second Generation managed package.
+    Lists any 1GP dependencies that are detected, as well as,
+    any dependency packages that have not benn promoted.
+    Once promoted, 2GP package can be installed into production orgs."""
+
+    task_options = {
+        "version_id": {
+            "description": "The SubscriberPackageVersion (04t) Id for the target package.",
+            "required": True,
+        },
+        "auto_promote": {
+            "description": (
+                "Automatically promote any unpromoted versions of dependency 2GP packages that are detected."
+            ),
+            "required": False,
+        },
+    }
+
+    def _init_options(self, kwargs) -> None:
+        super()._init_options(kwargs)
+
+        if "version_id" not in self.options:
+            raise TaskOptionsError("Task option `version_id` is required.")
```

Comment:
I think we probably need a way to find this based on the current commit so that it can be used from the Upload Release plan in MetaCI, where there's no place to enter a version_id.

Maybe something like: If the version_id isn't specified, look for a beta release tag that references the current commit, and use the version_id from its tag message. (Obviously in order for that to work we need to implement making the github_release task store the version_id in the message for the tag that it creates.)